mm2src/coins/solana.rs

```diff
@@ -57,15 +57,16 @@ use crate::{BalanceError, BalanceFut, CheckIfMyPaymentSentArgs, CoinFutSpawner,
             ValidateWatcherSpendInput, VerificationResult, WaitForHTLCTxSpendArgs, WatcherReward, WatcherRewardError,
             WatcherSearchForSwapTxSpendInput, WatcherValidatePaymentInput, WatcherValidateTakerFeeInput,
             WithdrawError, WithdrawFut, WithdrawRequest, WithdrawResult};
-
 pub mod solana_common;
 mod solana_decode_tx_helpers;
 pub mod spl;
 
-#[cfg(test)] mod solana_common_tests;
-#[cfg(test)] mod solana_tests;
+#[cfg(feature = "enable-solana")] pub mod solana_common_tests;
 #[cfg(test)] mod spl_tests;
 
+pub use solana_client;
+pub use solana_sdk;
```

Comment:
There is no need to make these mods and consts pub
![Screenshot 2024-08-08 at 14 12 18](https://github.com/user-attachments/assets/c7c8faa9-126e-4d51-b4b2-7917ecfee3bc)

Reply:
@laruh, I tried to access them from the Docker-related test crate, so I needed to expose them.

By the way, what kind of Clippy response is that? I can't seem to reproduce it. Could you provide some guidance on how to achieve that?
```bash
/komodo-defi-framework/: cargo clippy --all-targets --all-features -- -D warnings
warning: /komodo-defi-framework/mm2src/mm2_bin_lib/Cargo.toml: file `/komodo-defi-framework/mm2src/mm2_bin_lib/src/mm2_bin.rs` found to be present in multiple build targets:
  * `bin` target `mm2`
  * `bin` target `kdf`
   Compiling common v0.1.0 (/komodo-defi-framework/mm2src/common)
   Compiling ser_error v0.1.0 (/komodo-defi-framework/mm2src/derives/ser_error)
    Checking shared_ref_counter v0.1.0 (/komodo-defi-framework/mm2src/common/shared_ref_counter)
    Checking primitives v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/primitives)
    Checking test_helpers v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/test_helpers)
   Compiling serialization_derive v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/serialization_derive)
   Compiling trezor v0.1.1 (/komodo-defi-framework/mm2src/trezor)
   Compiling enum_derives v0.1.0 (/komodo-defi-framework/mm2src/derives/enum_derives)
    Checking mm2_number v0.1.0 (/komodo-defi-framework/mm2src/mm2_number)
    Checking serialization v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/serialization)
   Compiling ser_error_derive v0.1.0 (/komodo-defi-framework/mm2src/derives/ser_error_derive)
   Compiling coins v0.1.0 (/komodo-defi-framework/mm2src/coins)
    Checking bitcrypto v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/crypto)
    Checking mm2_state_machine v0.1.0 (/komodo-defi-framework/mm2src/mm2_state_machine)
    Checking keys v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/keys)
   Compiling mm2_main v0.1.0 (/komodo-defi-framework/mm2src/mm2_main)
   Compiling mm2_bin_lib v2.1.0-beta (/komodo-defi-framework/mm2src/mm2_bin_lib)
    Checking chain v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/chain)
    Checking script v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/script)
    Checking spv_validation v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/spv_validation)
    Checking rpc v0.1.0 (/komodo-defi-framework/mm2src/mm2_bitcoin/rpc)
    Checking mm2_metamask v0.1.0 (/komodo-defi-framework/mm2src/mm2_metamask)
    Checking mm2_db v0.1.0 (/komodo-defi-framework/mm2src/mm2_db)
    Checking mm2_err_handle v0.1.0 (/komodo-defi-framework/mm2src/mm2_err_handle)
    Checking mm2_event_stream v0.1.0 (/komodo-defi-framework/mm2src/mm2_event_stream)
    Checking db_common v0.1.0 (/komodo-defi-framework/mm2src/db_common)
    Checking mm2_p2p v0.1.0 (/komodo-defi-framework/mm2src/mm2_p2p)
    Checking mm2_metrics v0.1.0 (/komodo-defi-framework/mm2src/mm2_metrics)
    Checking hw_common v0.1.0 (/komodo-defi-framework/mm2src/hw_common)
    Checking rpc_task v0.1.0 (/komodo-defi-framework/mm2src/rpc_task)
    Checking mm2_io v0.1.0 (/komodo-defi-framework/mm2src/mm2_io)
    Checking mm2_rpc v0.1.0 (/komodo-defi-framework/mm2src/mm2_rpc)
    Checking mm2_eth v0.1.0 (/komodo-defi-framework/mm2src/mm2_eth)
    Checking mm2_core v0.1.0 (/komodo-defi-framework/mm2src/mm2_core)
    Checking mm2_gui_storage v0.1.0 (/komodo-defi-framework/mm2src/mm2_gui_storage)
    Checking crypto v1.0.0 (/komodo-defi-framework/mm2src/crypto)
    Checking utxo_signer v0.1.0 (/komodo-defi-framework/mm2src/coins/utxo_signer)
    Checking mm2_net v0.1.0 (/komodo-defi-framework/mm2src/mm2_net)
    Checking mm2_git v0.1.0 (/komodo-defi-framework/mm2src/mm2_git)
    Checking mm2_test_helpers v0.1.0 (/komodo-defi-framework/mm2src/mm2_test_helpers)
    Checking coins_activation v0.1.0 (/komodo-defi-framework/mm2src/coins_activation)
    Finished dev [unoptimized + debuginfo] target(s) in 1m 01s
warning: the following packages contain code that will be rejected by a future version of Rust: fs_extra v1.2.0, nom v6.1.2
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 3`
```

Reply:
> By the way, what kind of Clippy response is that? I can't seem to reproduce it. Could you provide some guidance on how to achieve that?

@mj-blockydevs I was using `cargo clippy --all-features -- -D warnings` command.
It's not the first time that `cargo clippy --all-features -- -D warnings` shows a warning that wasn't caught by `cargo clippy --all-targets --all-features -- -D warnings`.

hmm, clippy also doesn't detect unused function for me now.

```
~/RustroverProjects/komodo-defi-framework git:[solana-docker-tests-improvements-local-review]
cargo clippy --all-features -- -D warnings
warning: /Users/sharon/RustroverProjects/komodo-defi-framework/mm2src/mm2_bin_lib/Cargo.toml: file `/Users/sharon/RustroverProjects/komodo-defi-framework/mm2src/mm2_bin_lib/src/mm2_bin.rs` found to be present in multiple build targets:
  * `bin` target `mm2`
  * `bin` target `kdf`
   Compiling mm2_bin_lib v2.1.0-beta (/Users/sharon/RustroverProjects/komodo-defi-framework/mm2src/mm2_bin_lib)
    Finished dev [unoptimized + debuginfo] target(s) in 2.03s
warning: the following packages contain code that will be rejected by a future version of Rust: fs_extra v1.2.0, nom v6.1.2
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 622`
```

Well, you can search the project to see that it's not used
![Screenshot 2024-08-09 at 22 52 33](https://github.com/user-attachments/assets/99d7a22e-dfff-43a3-8f26-31bb4a0b2df4)

Sometimes it happens with clippy so I re check usage manually by searching in project files and checking usage in ide
![Screenshot 2024-08-09 at 22 58 51](https://github.com/user-attachments/assets/ded89416-04c9-49d6-b9c1-64fd78b554a6)

I recommend to add dead code annotation